06.10.2025/recursiveList.c

```diff
@@ -0,0 +1,185 @@
+#include "recursiveList.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+typedef struct ListNode {
+    int data;
+    struct ListNode* next;
+    struct ListNode* prev;
+} ListNode;
+
+/*
+ * Get the actual index of element.
+ */
+static int roundIndex(int idx, int listSize)
+{
+    idx = idx % listSize;
+    if (idx < 0)
+        idx = listSize + idx;
+    return idx;
+}
+
+static ListNode* allocListNode(int data, ListNode* next, ListNode* prev)
+{
+    ListNode* ptr = malloc(sizeof(struct ListNode));
+    if (ptr == NULL)
+        return ptr;
+
+    ptr->data = data;
+    ptr->next = next;
+    ptr->prev = prev;
+    return ptr;
+}
+
+/*
+ * Return the next node of deleted node.
+ * Node must be a node of pList.
+ */
+static ListNode* deleteNodeRecursiveList(RecursiveList* pList, ListNode* node)
+{
+    if (pList->_head == NULL)
+        return NULL;
+    if (node == pList->_head) {
+        if (node->next == node)
+            pList->_head = NULL;
+        else
+            pList->_head = pList->_head->next;
+    }
+
+    ListNode* prev = node->prev;
+    ListNode* next = node->next;
+    prev->next = next;
+    next->prev = prev;
+
+    pList->_size--;
+    free(node);
+    return pList->_head == NULL ? NULL : next;
+}
+
+/*
+ * Get node of the given index
+ */
+static ListNode* getNodeRecursiveList(const RecursiveList* pList, int idx)
+{
+    if (pList->_head == NULL)
+        return NULL;
+
+    ListNode* node = pList->_head;
+    for (int i = 0; i < idx; i++, node = node->next)
+        ;
```

Comment:
```suggestion
    for (int i = 0; i < idx; i++)
        node = node->next;
```

Давайте делать цикл `for` проще. И дальше по коду, если есть тоже.